gnovm/pkg/gnolang/machine.go

```diff
@@ -457,6 +459,41 @@ func (m *Machine) TestFunc(t *testing.T, tv TypedValue) {
 	})
 }
 
+// Stacktrace returns the stack trace of the machine.
+// It collects the executions and frames from the machine's frames and statements.
+func (m *Machine) Stacktrace() (stacktrace Stacktrace) {
+	if len(m.Frames) == 0 {
+		return
+	}
+
+	var calls []StacktraceCall
```

Comment:
```suggestion
	calls := make([]StacktraceCall, 0, len(m.Stmts))
```

I expect the total m.Frames to be generally 2-3 times as much compared to how many we'll need, so I think this is OK.